client/db_test.go

```diff
@@ -245,3 +245,29 @@ func ExampleDB_Insecure() {
 	// Output:
 	// aa=1
 }
+
+func ExampleDB_Open() {
```

Comment:
Yes, something like that is fine. I would get rid of the testFunc wrapper (just let the panic escape if it happens. I know it's a better error message if you recover from the panic but not worth the complexity). Rename testCase.err to testCase.expectErr and get rid of the `berr` variable (just do `if test.expectErr && err == nil { ... } else if !test.expectErr && err != nil {...}`)